DigitalLearningSolutions.Data/Models/Email/Email.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DigitalLearningSolutions.Data.Models.Email
+{
+    using MimeKit;
+
+    public class Email
+    {
+        public string[] To, Cc, Bcc;
+        public string Subject;
+        public BodyBuilder Body;
+
+        public Email()
+        {
+            this.To = new string[0];
+            this.Cc = new string[0];
+            this.Bcc = new string[0];
+            this.Subject = string.Empty;
+            this.Body = new BodyBuilder();
+        }
+
+        public Email(string? to, string? subject, BodyBuilder? body, string? cc = null, string? bcc = null)
+        {
+            this.To = (to is null) ? new string[0] : new string[1] {to};
+            this.Cc = (cc is null) ? new string[0] : new string[1] {cc};
+            this.Bcc = (bcc is null) ? new string[0] : new string[1] {bcc};
+            this.Subject = subject ?? string.Empty;
+            this.Body = body ?? new BodyBuilder();
+        }
+
+        public Email(string[]? to, string? subject, BodyBuilder? body, string[]? cc = null, string[]? bcc = null)
```

Comment:
Do you think it would be worth using default values of empty string array over null?

Reply:
I think it would, but after doing some research it looks like the only valid option is null as object references can't be used as compile-time constants (see: https://stackoverflow.com/questions/3480382/passing-an-empty-array-as-default-value-of-an-optional-parameter )

Reply:
Ah, fair! I think I've done too much Kotlin recently! 😄